WireAuthentication/Sources/WireAuthenticationUI/Views/DetermineAuthMethod/DetermineAuthMethodViewModel.swift

```diff
@@ -55,20 +57,52 @@ package final class DetermineAuthMethodViewModel: ObservableObject {
         !isValidEmailOrSSOCode()
     }
 
+    var isOnPremiseBackend: Bool {
+        environmentType != .production
+    }
+
     package init(
         router: any Router,
         factory: any Factory,
         environmentType: BackendEnvironmentType,
         backendConfig: BackendConfig,
+        backendMetadata: BackendMetadata?,
         emailOrSSOCode: String = "",
-        isLoading: Bool = false
+        isLoading: Bool = false,
+        bridge: WireAuthenticationBridge
     ) {
         self.router = router
         self.factory = factory
         self.environmentType = environmentType
+        self.backendMetadata = backendMetadata
         self.backendConfig = backendConfig
         self.emailOrSSOCode = emailOrSSOCode
         self.isLoading = isLoading
+
+        self.cancellable = bridge.inboundEvents.sink { event in
+            switch event {
+            case let .onSwitchBackend(configURL):
+                Task {
+                    let resolvedBackendMetadata: BackendMetadata
```

Comment:
suggestion: I understand that we resolve the metadata here so we can invoke `handleAuthenticationMethod`, but we pass `.onPremLogin` which doesn't use the backendMetadata. We could avoid this if we factor out the handling of `.onPremLogin` into a private method and just invoke that directly. What do you think?

Reply:
Yes, it makes sense not to introduce additional requests.
Improved